Feat: #100 NumberTextField

wish627 opened the pull request:
## ✨ 작업 내용
### 📌 관련 이슈 (Related Issue)
<!-- 해당 PR이 어떤 이슈를 해결하는지 연결해주세요 -->
- Closes #100

### 🧶 주요 변경 내용 (Summary)
<!-- 이번 PR에서 작업한 핵심 변경 사항을 작성해주세요 -->

- 숫자만 입력받는 텍스트필드
- 사용하지 않는 코드 삭제
- 토글에서 글자 외의 구역도 클릭 가능하게 수정

---

### 📸 스크린샷 (Optional)
<!-- UI 작업의 경우, 구현한 화면을 첨부해주세요 -->
<!-- 이미지 크기 조절 예시: <img width="300" alt="설명" src="링크"> -->


---

## ✅ 체크리스트

예시)
- [x] 관련 이슈를 닫는 커밋 메시지 사용 (`Closes #`)
- [x] 동작 확인 완료 (시뮬레이터 / 실기기)
- [x] PR 제목, 설명 명확히 작성됨
- [x] 커밋 메시지 규칙 준수
- [x] 불필요한 주석/코드 제거

---

## 💬 리뷰 포인트
<!-- 로직 고민, UI 구성, 네이밍 등 의견 요청 -->
별 거 없는 내용이지만 PR 올리고 버튼 컴포넌트로 disable 처리하러 갑니다

wish627 committed:
feat: #100 시간 범위 제한

10분에서 30분 사이의 시간

wish627 committed:
chore: #100 사용하지 않는 코드 정리

불필요한 numberFormatter 제거, .background(Color.clear) 생략

wish627 committed:
feat: #100 ToggleTabView 선택 영역 확장

텍스트 외 영역 선택시에도 토글 동작

ohdodin commented on NoMyeolmang/NoMyeolmang/Sources/Presentation/TimerSettingView/View/PersonalTimerSettingView.swift:
```diff
@@ -40,24 +32,22 @@ struct PersonalTimerSettingView: View {
                             }
                             // goalTime에 직접 연결
-                            if let time = Int(filtered), time > 0 {
+                            if let time = Int(filtered), (10...30).contains(time) {
```
(10...30).contains(time)에서 도출되는 bool 값이 여기 뿐만 아니라 disable에도 쓰일 거 같아서 함수로 만들어주세요!

wish627 replied:
확인 후 이행 완 !

ohdodin replied:
적용이 덜 되었지만... 다음 disable 이슈에서 적용하는 전제하에 통과 드리겠습니다

ohdodin reviewed (commented):
굳 확인완이고 아래 수정사항 하나만 바꿔주세요!

wish627 committed:
feat: #100 isTimeValid 함수화

(10...30).contains(time)에서 도출되는 bool 값이 여기 뿐만 아니라 disable에도 쓰일 거 같아서 함수로 제작 (도딘의 리뷰)

ohdodin reviewed (approved):
일단 통과 드립니돠
